### PR TITLE
chore(Menu): convert examples to TypeScript

### LINKED ISSUE #7570: Menu: convert examples to TypeScript

### PR DESCRIPTION
**Pull request by jenny-s51**
**What**: Closes #7570

### CODE REVIEW

**Review by thatblindgeye — COMMENTED** [review]
Looking good so far! Had some comments for issues I noticed below, as well as the following:

- For most of the examples, the `onSelect` method doesn't have any types on the parameters like the MenuWithCheckbox file does.

**Review by jenny-s51 — COMMENTED** [review]
Thank you for your review @thatblindgeye! 

Addressed most of your comments. I'm investigating the focus issue you mentioned, which doesn't have a trivial solution. 

The TSX/React Hooks version of the `setState` callback is `useEffect`, but combining `useEffect` with `refs` doesn't work the way we'd expect, so I will likely have to rewrite some logic in the `ViewMore` demo. Currently a WIP!

File: packages/react-core/src/components/Menu/examples/MenuWithDrilldownBreadcrumbs.tsx [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+  BadgeToggle,
+  Checkbox
+} from '@patternfly/react-core';
+import StorageDomainIcon from '@patternfly/react-icons/dist/esm/icons/storage-domain-icon';
+import CodeBranchIcon from '@patternfly/react-icons/dist/esm/icons/code-branch-icon';
+import LayerGroupIcon from '@patternfly/react-icons/dist/esm/icons/layer-group-icon';
+import AngleLeftIcon from '@patternfly/react-icons/dist/esm/icons/angle-left-icon';
+import CubeIcon from '@patternfly/react-icons/dist/esm/icons/cube-icon';
+
+export const MenuWithDrilldownBreadcrumbs: React.FunctionComponent = () => {
```
[the remark this answers is not in the document]
Comment (jenny-s51) [reply packages/react-core/src/components/Menu/examples/MenuWithDrilldownBreadcrumbs.tsx:24]: Nice catch - fixed!

**Comment by thatblindgeye** [comment]
Per discussion with @jenny-s51, I pushed a potential fix to the issue that arose with the View More example after converting to TS/functional components. One area of concern might be if the `activeItem` is set to a valid item when initialized (rather than `0`/an invalid item in the example), but that could be resolved by passing a conditional into the useEffect hook to only set focus when the `numOptions` is greater than 3/the initial amount.

**Review by kmcfaul — APPROVED** [review]
Looks good!

There is a bit of a keyboard issue with drilldown + breadcrumbs I'm noticing where shift tabbing while within a drilled in menu won't immediately take focus to the breadcrumbs (requiring multiple shift tab keypresses). It looks like this was also present in our current docs so it wasn't the conversion, I believe it's probably to do with drilldown replacing the active menu in place so it must shift tab through previous menu containers before hitting the breadcrumbs. Since it's not a new bug I think it's probably outside this PR scope and should be a follow up/separate issue.

**Review by jenny-s51 — COMMENTED** [review]
Thank you for adding back your updates @thatblindgeye !

@kmcfaul Ah nice catch - definitely a weird bug! I'm seeing that behavior persist in the existing docs as well. Since you're the one who caught this, would you like to open the related issue?

**Review by jenny-s51 — COMMENTED** [review]
Conflicts should now be resolved - did a big rebase to pull in @nicolethoen 's updates from https://github.com/patternfly/patternfly-react/pull/7604/files

**Comment by kmcfaul** [comment]
@jenny-s51 Sure will do

**Review by nicolethoen — APPROVED** [review]

**Pull request merged**